### DISCUSSION TITLE
Query: How can i load a custom XDP program using CNDP

### DISCUSSION THREAD

**Discussion by Lakshyagour** [CloudNativeDataPlane/cndp#338] [Q&A] [answered]
I have written a custom XDP program to filter packets for application. I made changes in the cndp code to load my program instead but I am facing a segmentation fault when trying to add entry into the map structure.

**Comment by maryamtahhan** [comment, marked as answer]
Hi @Lakshyagour 
I have a patch here that I intend to clean up and make a contribution to CNDP that you can try out: https://github.com/maryamtahhan/veth-benchmark/blob/main/containerization/0001-Load-custom-bpf-program.patch you can apply it on top of git commit: 53f60b170bc320ad87c5b2f8c932e88cd2dc4370 and see if it helps out or to spot any issues with your own patch...
I have used it with this bpf filtering propgram https://github.com/maryamtahhan/veth-benchmark/tree/main/containerization/filter-udp-to-af_xdp

**Reply by Lakshyagour** [reply]
Thank You.

**Comment by maryamtahhan** [comment]
@Lakshyagour You can also use the xdp map pinning support in cndp.

https://github.com/CloudNativeDataPlane/cndp/pull/283

An example is in the cndp-frr example. We use xdp-loader to load and pin maps then pass the pin paths to CNDP.
